demosplan/DemosPlanCoreBundle/Logic/FileResponseGenerator/ZipResponseGenerator.php

```diff
@@ -0,0 +1,178 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of the package demosplan.
+ *
+ * (c) 2010-present DEMOS plan GmbH, for more information see the license file.
+ *
+ * All rights reserved
+ */
+
+namespace demosplan\DemosPlanCoreBundle\Logic\FileResponseGenerator;
+
+use demosplan\DemosPlanCoreBundle\Entity\File;
+use demosplan\DemosPlanCoreBundle\Exception\AssessmentTableZipExportException;
+use demosplan\DemosPlanCoreBundle\Exception\InvalidDataException;
+use demosplan\DemosPlanCoreBundle\Logic\Procedure\NameGenerator;
+use demosplan\DemosPlanCoreBundle\Logic\ZipExportService;
+use demosplan\DemosPlanCoreBundle\Utilities\DemosPlanPath;
+use Exception;
+use InvalidArgumentException;
+use PhpOffice\PhpSpreadsheet\Writer\Exception as WriterException;
+use PhpOffice\PhpSpreadsheet\Writer\IWriter;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\StreamedResponse;
+use Symfony\Contracts\Translation\TranslatorInterface;
+use Webmozart\Assert\Assert;
+use ZipStream\Exception\FileNotFoundException;
+use ZipStream\Exception\FileNotReadableException;
+use ZipStream\ZipStream;
+
+class ZipResponseGenerator extends FileResponseGeneratorAbstract
+{
+    private const FIILE_NOT_FOUND_OR_READABLE = 'Unable to load or read file from path.';
+    private const FILE_HASH_INVALID = 'The hash for a file, which should be added to a zip export is not a string,
```

Comment:
Why do you mix transkeys and plain messages?

Reply:
My last information was that log messages should not be created as translation keys.

Reply:
ok, fair enough, this is correct